local/cs2050-cpu.yml.erb

```diff
@@ -0,0 +1,67 @@
+# Custom setup for COMPSCI 2050. Allows users to set a number of CPU cores up to
+# the total number of cores available on our general purpose compute nodes,
+# currently 36. App is named to indicate it is a CPU app, since CS 2050 has
+# requested the same setup in the GPU queue.
+
+# Included in initial template
+<%-
+  groups = OodSupport::User.new.groups.sort_by(&:id).tap { |groups|
+    groups.unshift(groups.delete(OodSupport::Process.group))
+  }.map(&:name).grep(/^P./)
+-%>
+# Added to control group access to apps
+<%-
+def arrays_have_common_element(array1, array2)
+  # Use the `&` operator to get the intersection of the two arrays
+  # If the intersection is not empty, return true, otherwise false
+  !(array1 & array2).empty?
+end
+
+userGroups = OodSupport::User.new.groups.sort_by(&:id).map(&:name)
+# First check if the user is in an admin group
+adminGroups = [
+  "ondemand-admins-1025174" # HUIT OOD admin group, prod environment
+]
+if arrays_have_common_element(userGroups, adminGroups)
+  cluster="*"
+else
+  # If the user is not in an admin group, check if they're in an authorized Canvas group
+  userCanvasGroups = userGroups.flat_map{ |str| str.scan(/^canvas(\d+)-\d+/) }.flatten
+  enabledGroups = [
+    # Cannot have other enabled groups if a course installation is in use
+    # This is because the course installation uses the course shared folder,
+    # which is not accessible to users outside of that course.
+    # "144286" # COMPSCI 2050: High Performance Computing for Science and Engineering
+  ]
+
+  # Check if the groups that the user is in match any of the courses that should
+  # have access to this app.
+
+  if arrays_have_common_element(userCanvasGroups, enabledGroups)
+    cluster="*"
+  else
+    cluster="disable_this_app"
+  end
+end
+-%>
+---
+title: Code Server (COMPSCI 2050 - CPU)
+cluster: "<%= cluster %>"
+cacheable: false
+form:
+  - bc_num_hours
+  - bc_num_slots
+  - bc_queue
+  - version
+  - custom_num_cores
+attributes:
+  version: "4.12.0"
```

Comment:
Have there been any requests to upgrade Code Server to a more recent version? I noticed that [v4.12.0](https://github.com/coder/code-server/releases/tag/v4.19.0) was released in Nov 2023 and they're up to [v4.99.1](https://github.com/coder/code-server/releases/tag/v4.99.1), which was released just a few days ago. At this point in the semester, it seems like we could just wait to upgrade.

Reply:
We haven't had any requests for it, but I'd be interested in upgrading. This was the most recent version available on Spack at the time, but I just checked and it looks like [4.96](https://packages.spack.io/package.html?name=code-server) is available now, so that should make it pretty easy to upgrade.